web-ui/src/app/components/address-details/address-details.component.html

```diff
@@ -4,57 +4,79 @@
-            </tr>
-            <tr>
-              <td>{{'label.received' | translate}}</td>
-              <td>{{address.received}} {{'label.coinName' | translate}}</td>
-            </tr>
-            <tr>
-              <td>{{'label.spent' | translate}}</td>
-              <td>{{getSent(address)}} {{'label.coinName' | translate}}</td>
-            </tr>
-          </tbody>
-        </table>
+            <tbody>
+              <tr>
+                <td>{{'label.address' | translate}}</td>
+                <td>{{addressString}}</td>
+              </tr>
+              <tr>
+                <td>{{'label.balance' | translate}}</td>
+                <td>{{address.available}} {{'label.coinName' | translate}}</td>
+              </tr>
+              <tr>
+                <td>{{'label.received' | translate}}</td>
+                <td>{{address.received}} {{'label.coinName' | translate}}</td>
+              </tr>
+              <tr>
+                <td>{{'label.spent' | translate}}</td>
+                <td>{{getSent(address)}} {{'label.coinName' | translate}}</td>
+              </tr>
+            </tbody>
+          </table>
+        </div>
       </div>
     </div>
 
     <div class="row">
-      <div class="col-xs-12 col-sm-12 col-md-4 col-lg-4">
-        <table class="table table-condensed table-bordered table-striped table-hover">
-          <thead>
-            <tr>
-              <th class="col-xs-2 col-sm-2 col-md-1 col-lg-1">{{'label.transactions' | translate}} ({{address.transactions.length}})</th>
-            </tr>
-          </thead>
+      <h2 class="col-xs-12">{{'label.transactions' | translate}}</h2>
+    </div>
+    <div class="table-responsive">
+      <table class="table table-condensed table-bordered table-striped table-hover">
+        <thead>
+          <tr>
+            <th class="col-xs-1">#</th>
+            <th class="col-xs-2">{{'label.transaction' | translate}}</th>
+            <th class="col-xs-2">{{'label.blockhash' | translate}}</th>
+            <th class="col-xs-2">{{'label.date' | translate}}</th>
+            <th class="col-xs-3">{{'label.value' | translate}}</th>
+            <th class="col-xs-2">{{'label.size' | translate}}</th>
+          </tr>
+        </thead>
+
+        <tbody>
+          <tr *ngFor="let index = index; let item of asyncItems | async | paginate: { id: 'transactions', itemsPerPage: pageSize, currentPage: currentPage, totalItems: total }">
+            <td>{{(currentPage - 1) * pageSize + index + 1}}</td>
+            <td>
+              <a routerLink="/transactions/{{item.id}}">{{item.id | slice:0:15}}...</a>
+            </td>
+            <td>
+              <a routerLink="/blocks/{{item.blockhash}}">{{item.blockhash | slice:0:15}}...</a>
+            </td>
+            <td>{{item.time * 1000 | date:'MMMM d, y, h:mm:ss a'}}</td>
+            <td>{{item.received >= item.sent ? '+' : ''}}{{item.received - item.sent}} {{'label.coinName' | translate}}</td>
+            <td>{{item.size}}</td>
```

Comment:
If you see the explorer, size is displayed in bytes: https://xsnexplorer.io/blocks/7f6645679c9ad7388092f749a5f62012daef554557fe428466933e37bdfe2115